### ISSUE TITLE
the contenthash is different

### ISSUE THREAD

**Issue by lubyxu** [webpack-contrib/mini-css-extract-plugin#75]
the plugin is updated to the latest version. however,  the same code is bundled twice, the contenthashes produced each time are different.

**Comment by evilebottnawi** [comment]
@lubyxu can you create minimum reproducible test repo?

**Comment by lubyxu** [comment]
In the case of the config:
```
output: {
    filename: '[name].[hash].bundle.js',
    publicPath: '/dist/'
},
externals: {
    react: 'react'
},
plugins: [
    new MiniCssExtractPlugin({
        filename: '[name].[contenthash].css',
        chunkFilename: '[id].[contenthash].css'
    })
],
optimization: {
    splitChunks: {
        cacheGroups: {
            commons: {
                test: /\.css/,
                name: 'commons',
                chunks: 'initial',
                minChunks: 2,
                minSize: 1
            }
        }
    },
}
```
The outputs of bundle twice as followes:
The outputs of the first bundle:
![image](https://user-images.githubusercontent.com/8433782/38189500-1c5c0de4-3693-11e8-8bcd-716b4edc9260.png)

The outputs of the second bundle:
![image](https://user-images.githubusercontent.com/8433782/38189516-334ab050-3693-11e8-95a9-08b511b853fa.png)

Compared with the results above, the `d500b09e2d2fa5327bea` contenthash produced by the plugin are the same. However, the `ea0c09dc18776505059c` hash produced by webpack are different.

When I remove the mini-css-extract-plugin. the results are the same as below:
The outputs of the first bundle: 
![image](https://user-images.githubusercontent.com/8433782/38189529-480126aa-3693-11e8-9529-44f0452c7570.png)

The outputs of the second bundle:
![image](https://user-images.githubusercontent.com/8433782/38189540-52fda826-3693-11e8-9e9b-1496ac9a3c1c.png)

The link is: https://github.com/lubyxu/mini-css-extract-plugin-case.

Thank u for your reply

**Comment by evilebottnawi** [comment]
@lubyxu PR https://github.com/webpack/webpack/pull/6943

**Comment by egemon** [comment]
I just spent two days trying to fix the same issue.
Two subsequent builds of the same code produces the same [contenthash] for extracted css and produces different  [contenthash] almost for all .js files. 
Finally fixed it upgrading mini-css-extract-plugin from v0.4.4 to v0.6.0
I come to it just with huge amount of experiments and testing.

webpack v4.29.6
mini-css-extract-plugin v0.4.4

I will put here all phrases that I googled trying to solve this, maybe smbd will find this here, as it is also about the same issue:
MiniCssExtractPlugin modify contenthash of js on rebuild
webpack inconsistent contenthash
HashedModuleIdsPlugin doesn't work
webpack contenthash changes while content not changing
chunks contenthash change on each build
